service/src/test/kotlin/no/nav/su/se/bakover/service/behandling/UnderkjennBehandlingTest.kt

```diff
@@ -473,4 +485,185 @@ class UnderkjennBehandlingTest {
             behandlingMetricsMock,
         )
     }
+
+    @Test
+    fun `kan oppdatere behandlingsinformasjon med status UNDERKJENT`() {
+        val behandling: Behandling =
+            innvilgetBehandlingTilAttestering.copy(status = Behandling.BehandlingsStatus.UNDERKJENT)
+
+        val behandlingRepoMock = mock<BehandlingRepo> {
+            on { hentBehandling(any()) } doReturn behandling
+        }
+
+        val actual = BehandlingTestUtils.createService(
+            behandlingRepo = behandlingRepoMock,
+            microsoftGraphApiOppslag = BehandlingTestUtils.microsoftGraphMock.oppslagMock
+        ).oppdaterBehandlingsinformasjon(
+            behandlingId = behandling.id,
+            saksbehandler = saksbehandler,
+            behandlingsinformasjon = Behandlingsinformasjon.lagTomBehandlingsinformasjon().copy(
+                uførhet = Behandlingsinformasjon.Uførhet(
+                    status = Behandlingsinformasjon.Uførhet.Status.VilkårIkkeOppfylt,
+                    uføregrad = null,
+                    forventetInntekt = null,
+                    begrunnelse = null
+                ),
+                flyktning = Behandlingsinformasjon.Flyktning(
+                    status = Behandlingsinformasjon.Flyktning.Status.VilkårIkkeOppfylt,
+                    begrunnelse = null
+                )
+            )
+        )
+
+        actual shouldBe behandling.copy(
+            status = Behandling.BehandlingsStatus.VILKÅRSVURDERT_AVSLAG,
+            behandlingsinformasjon = Behandlingsinformasjon.lagTomBehandlingsinformasjon().copy(
+                uførhet = Behandlingsinformasjon.Uførhet(
+                    status = Behandlingsinformasjon.Uførhet.Status.VilkårIkkeOppfylt,
+                    uføregrad = null,
+                    forventetInntekt = null,
+                    begrunnelse = null
+                ),
+                flyktning = Behandlingsinformasjon.Flyktning(
+                    status = Behandlingsinformasjon.Flyktning.Status.VilkårIkkeOppfylt,
+                    begrunnelse = null
+                )
+            )
+        ).right()
+
+        verify(behandlingRepoMock).hentBehandling(argThat { it shouldBe behandling.id })
+    }
+
+    @Test
+    fun `kan lage ny beregning med status UNDERKJENT`() {
```

Comment:
Hva skjer hvis man forsøker å gjøre dette for en behandling som i utgangspunktet var avslått pga vilkår? Ref tidligere kommentarer ang "chain-of-events".